File: clojure/gigasecond/src/gigasecond.clj
(ns gigasecond
  (:import (java.time LocalDateTime Duration)))

(defn exp [x n]
  (reduce * (repeat n x)))

(def ^:private gigaseconds-duration (Duration/ofSeconds (exp 10 9)))

(defn format-output [date]
  [(.getYear date) (.getMonthValue date) (.getDayOfMonth date)])

(defn from [year month day]
  (-> (LocalDateTime/of year month day 0 0)
      (.plus gigaseconds-duration)
      format-output))

File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.math.BigInteger
import java.time.LocalDate
import java.time.LocalDateTime

/**
 * Created by yannickgrenzinger on 12/08/2016.
 */
class Gigasecond(birthDate: LocalDateTime) {
    val gigaseconds = BigInteger.TEN.pow(9).toLong()
    val date: LocalDateTime = birthDate.plusSeconds(gigaseconds)

    constructor(birthDate: LocalDate) : this(birthDate.atStartOfDay())
}